src/main/java/org/jabref/logic/importer/fileformat/PdfMergeMetadataImporter.java

```diff
@@ -161,17 +161,67 @@ private void fetchData(BibEntry candidate, StandardField field, IdBasedFetcher f
     }
 
     private static BibEntry mergeCandidates(Stream<BibEntry> candidates) {
-        final BibEntry entry = new BibEntry();
-        candidates.forEach(entry::mergeWith);
+        // Convert the stream to a list so we can iterate over the list twice
+        List<BibEntry> candidateList = candidates.toList();
 
-        // Retain online links only
-        List<LinkedFile> onlineLinks = entry.getFiles().stream().filter(LinkedFile::isOnlineLink).toList();
+
+        BibEntry entry = new BibEntry();
+
+        //  Score titles to find the "best" title among candidates
+        int bestTitleScore = -1;
+        String bestTitle = null;
+
+        for (BibEntry candidate : candidateList) {
+            Optional<String> candidateTitle = candidate.getField(StandardField.TITLE);
+            if (candidateTitle.isPresent()) {
+                int score = calculateTitleScore(candidateTitle.get());
+                if (score > bestTitleScore) {
+                    bestTitleScore = score;
+                    bestTitle = candidateTitle.get();
+                }
+            }
+        }
+
+        // Merge all fields from the candidates, same as previous method
+        candidateList.forEach(entry::mergeWith);
+
+        // Override the best title we found
+        if (bestTitle != null) {
+            entry.setField(StandardField.TITLE, bestTitle);
+        }
+
+
+        List<LinkedFile> onlineLinks = entry.getFiles().stream()
+                .filter(LinkedFile::isOnlineLink)
+                .toList();
         entry.clearField(StandardField.FILE);
         entry.addFiles(onlineLinks);
 
         return entry;
     }
 
+    private static int calculateTitleScore(String title) {
+        //for every word in the title, plus one point
+        int wordCount = title.trim().split("\\s+").length;
+        if(wordcount > 35){
+            wordcount = -2; //super long titles are less favourable
+        }
+
+        //if the title ends in .ccc or .cccc where c is any alphabetic char, minus 10 points
+        int endsInExtension= title.matches(".*\\.[a-zA-Z]{3,4}") ? -10 : 0;
+
+        int endsWithFileExtension = 0;
+
+        if (title.matches("(?i).*(\\.(pdf|docx?|txt|jpg|png))$")){
```

Comment:
```suggestion
        if (title.matches("(?i).*(\\.(pdf|docx|odt|txt|jpg|png))$")){
```
Is the `?` a typo?

Reply:
RegEx.

Pattern.compile should be used

Reply:
What I mean: docx? matches both doc and docx, which are both valid Word extensions.

However, this functionality is not needed here. Only in very, very seldom cases, titles contain file names. -- Maybe the students found some interesting cases; but then hey should provide test cases